src/components/virtualRepeat/virtualRepeater.js

```diff
@@ -665,6 +702,28 @@ VirtualRepeatController.prototype.updateIndexes_ = function() {
   this.newStartIndex = Math.max(0, Math.min(
       itemsLength - containerLength,
       Math.floor(this.container.getScrollOffset() / this.itemSize)));
-  this.newEndIndex = Math.min(itemsLength, this.newStartIndex + containerLength + NUM_EXTRA);
+  this.newVisibleEnd = this.newStartIndex + containerLength + NUM_EXTRA;
+  this.newEndIndex = Math.min(itemsLength, this.newVisibleEnd);
   this.newStartIndex = Math.max(0, this.newStartIndex - NUM_EXTRA);
 };
+
+
+function VirtualRepeatModelArrayLike(model) {
+  this.model = model;
+  this.length = model.getLength();
```

Comment:
Hmm... I'd rather not have to do these checks on each digest cycle, but for full correctness I'd have to.